vowpalwabbit/search.cc

```diff
@@ -1027,7 +1016,8 @@ action single_prediction_notLDF(search_private& priv, example& ec, int policy, c
 
   cdbg << "allowed_actions_cnt=" << allowed_actions_cnt << ", ec.l = ["; for (size_t i=0; i<ec.l.cs.costs.size(); i++) cdbg << ' ' << ec.l.cs.costs[i].class_index << ':' << ec.l.cs.costs[i].x; cdbg << " ]" << endl;
 
-  priv.base_learner->predict(ec, policy);
+  as_singleline(priv.base_learner)->predict(ec, policy);
```

Comment:
Why can't we store whether or not something is singleline in the base_learner?

Reply:
todo: shift earlier.

Reply:
In search priv.base_learner can be either a single-line or a multi-line depending on command line arguments.